lld/ELF/Writer.cpp

```diff
@@ -2465,13 +2460,22 @@ SmallVector<PhdrEntry *, 0> Writer<ELFT>::createPhdrs(Partition &part) {
     // so when hasSectionsCommand, since we cannot introduce the extra alignment
     // needed to create a new LOAD)
     uint64_t newFlags = computeFlags(sec->getPhdrFlags());
+    // When --no-rosegment is specified, RO and RX sections are compatible.
+    uint32_t diff = flags ^ newFlags;
+    if (config->singleRoRx && !(newFlags & PF_W))
+      diff &= ~PF_X;
```

Comment:
There's an interesting conflict here between singleRoRx and execute-only, could be worth making these options incompatible in Driver.cpp as singleRoRx breaks execute-only.

This isn't a regression caused by this patch though so could be done in a different patch.

Reply:
The initial `--execute-only` patch (https://reviews.llvm.org/D49456) has an error `--execute-only does not support intermingling data and code`. 

It also has this error `"--execute-only and -no-rosegment cannot be used together"`. (The `!Script->HasSectionsCommand` condition it uses should be wrong as hasSectionsCommand is not set in Driver.cpp yet)